packages/expo-updates/src/UseUpdates.ts

```diff
@@ -64,9 +64,14 @@ export const useUpdates: () => UseUpdatesReturnType = () => {
 
   // Change the state based on native state machine context changes
   useEffect(() => {
-    getNativeStateMachineContextAsync().then((context) => {
-      setUpdatesState((updatesState) => reduceUpdatesStateFromContext(updatesState, context));
-    });
+    getNativeStateMachineContextAsync()
+      .then((context) => {
+        setUpdatesState((updatesState) => reduceUpdatesStateFromContext(updatesState, context));
+      })
+      .catch((error) => {
+        // Native call can fail (e.g. if in development mode), so catch the promise rejection and surface the error
+        setUpdatesState((updatesState) => ({ ...updatesState, checkError: error }));
+      });
     const subscription = addUpdatesStateChangeListener((event) => {
```

Comment:
should we only add the subscription if the initialization succeeds?

Reply:
It's certainly safe to go ahead and add the subscription.